### ISSUE TITLE
sql: disallow some schema changes in transactions

### ISSUE THREAD

**Issue by jseldess** [cockroachdb/docs#1273]
PR: https://github.com/cockroachdb/cockroach/pull/14368

**Comment by andreimatei** [comment]
note that some of https://github.com/cockroachdb/cockroach/pull/14368 has been rolled back in  https://github.com/cockroachdb/cockroach/pull/15511

**Comment by jseldess** [comment]
@vivekmenezes, can you summarize the change here? We have this [known limitation](https://www.cockroachlabs.com/docs/dev/known-limitations.html#schema-changes-within-transactions). Does it need to be updated?

**Comment by vivekmenezes** [comment]
" This also applies to running TRUNCATE on such a table because TRUNCATE implicitly drops and recreates the table" should be removed. I can take it on for doc fixit day whenever that is! We should do more of them!

**Comment by jseldess** [comment]
Great. It'll be next Thursday!
